google/cloud/spanner/internal/build_info.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_INTERNAL_BUILD_INFO_H_
+#define GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_INTERNAL_BUILD_INFO_H_
+
+#include "google/cloud/spanner/version.h"
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+namespace internal {
+
+/**
+ * Returns the compiler ID.
+ *
+ * The Compiler ID is a string like "GNU" or "Clang", as described by
+ * https://cmake.org/cmake/help/v3.5/variable/CMAKE_LANG_COMPILER_ID.html
+ */
+std::string CompilerId();
+
+/**
+ * Returns the compiler version.
+ *
+ * This string will be something like "9.1.1".
+ */
+std::string CompilerVersion();
+
+/**
+ * Returns the compiler flags.
+ *
+ * Examples include "-c fastbuild" or "-O2 -DNDEBUG".
```

Comment:
Is `-c fastbuild` really a compiler flag?

Reply:
>> Is -c fastbuild really a compiler flag?
>
> We don't have the actual compiler flags that were used ... so we use the COMPILATION_MODE as a proxy.

OK.  Then I'll only suggest that something like `BuildFlags()` might be a better name.